test/async_cache_test.dart

```diff
@@ -158,4 +158,20 @@ void main() {
     }));
     expect(cache.fetchStream(call).toList(), completion(['1', '2', '3']));
   });
+
+  test('should invalidate even if the future throws an exception', () async {
+    cache = AsyncCache.ephemeral();
+
+    Future<String> throwingCall() async => throw Exception();
+    try {
+      await cache.fetch(throwingCall);
+    } catch (exception) {
+      expect(exception, isException);
+    }
+    // To let the timer invalidate the cache
+    await Future.delayed(Duration.zero);
```

Comment:
This is probably safe. I'd consider using a non-null duration, just to be absolutely sure that it triggers later than the other zero-duration timer, even if they are scheduled within the same millisecond.
(Just to be safe, probably not an issue in practice).

Reply:
I've added 5 milliseconds to the duration.